internal/helm/release/manager.go

```diff
@@ -77,7 +77,7 @@ type manager struct {
 }
 
 type InstallOption func(*action.Install) error
-type UpgradeOption func(*action.Upgrade) error
+type UpgradeOption func(UpgradeConfig) error
```

Comment:
Why are we reverting this? We can directly use `action.Upgrade` and `action.Rollback` (https://github.com/helm/helm/blob/main/pkg/action/rollback.go#L35) from helm config, right?

Reply:
The rollback handler is initialized within the `UpgradeRelease` method, specifically in line 219. 

 https://github.com/operator-framework/operator-sdk/blob/e67da35ef4fff3e471a208904b2a142b27ae32b1/internal/helm/release/manager.go#L206-L235 .

Given `action.Rollback` is not exposed as a parameter to the method, it seemed using the `UpgradeConfig` would allow it to be configured without changing the `Manager` interface  in line 50. See below:

https://github.com/operator-framework/operator-sdk/blob/e67da35ef4fff3e471a208904b2a142b27ae32b1/internal/helm/release/manager.go#L50-L60 

Let me know if I am missing anything.

Reply:
The `Manager` interface is not user-facing as of now, so its safe to make changes to it. Would this make it better:
Introduce a rollback method in manager's interface:
```go
RollBack(context.Context, RollBackOption) (*rpb.Release, *rpb.Release, error)
```

Have this call `action.NewRollback`. And call this when upgrade is unsuccessful - as in [here](https://github.com/operator-framework/operator-sdk/pull/6546/files#diff-e75c2e3a252b9563505b262b8053234e75da15768a1fdbc7d4daa09cc07a3a90R234-R235). This would make it easier to later customize the rollback method specific to SDK. Wdyt?

Reply:
Roger that. I will explore the suggestion and push a PR if it works.

Reply:
@varshaprasad96 For the RollBack method to be called within the `UpgradeRelease` method, we need to be able to pass down the force rollback value via the `UpgradeRelease` method.

The only other workaround is to evaluate the CR annotation for forced rollback when creating a new Manager object using the `NewManager` factory method. Which means annotations would be evaluated in two different locations.

Reply:
Hold on. I believe if we remove the rollback out of the `UpgradeRelease` method and return an error whenever an upgrade fails. Then we should be able to customize the behavior of the rollback to our liking. I will  explore this on tomorrow. This may require rewriting tests as well.